src/benchq/resource_estimators/resource_info.py

```diff
@@ -26,26 +28,51 @@ class DecoderInfo:
 
 
 @dataclass
-class DetailedIonTrapResourceInfo:
+class MagicStateFactoryInfo:
```

Comment:
To avoid redundancy, let's delete the `MagicStateFactory` class and everywhere it is used, instead use `MagicStateFactoryInfo`.

Reply:
Done!